Error compiling with websockets

This is when compiling with boost 1.70

> /home/jonesmz/mqtt_cpp/include/mqtt/server.hpp:514:33: error: ‘using element_type = class mqtt::ws_endpoint<boost::asio::basic_stream_socket<boost::asio::ip::tcp>, boost::asio::io_context::strand>’ {aka ‘class mqtt::ws_endpoint<boost::asio::basic_stream_socket<boost::asio::ip::tcp>, boost::asio::io_context::strand>’} has no member named ‘lowest_layer’; did you mean ‘next_layer’?
>                          socket->lowest_layer().close(close_ec);
>                          ~~~~~~~~^~~~~~~~~~~~
>                          next_layer

--- a/include/mqtt/ws_endpoint.hpp
+++ b/include/mqtt/ws_endpoint.hpp
@@ -45,9 +45,19 @@ public:
         return ws_.get_io_service();
     }
 
+#if BOOST_VERSION >= 107000
+
+    auto& lowest_layer() {
+        return boost::beast::get_lowest_layer(ws_);
+    }
+
+#else  // BOOST_VERSION >= 107000
+
     typename boost::beast::websocket::stream<Socket>::lowest_layer_type& lowest_layer() {
         return ws_.lowest_layer();
     }
+
+#endif // BOOST_VERSION >= 107000
 
     typename boost::beast::websocket::stream<Socket>::next_layer_type& next_layer() {
         return ws_.next_layer();

